Tests/SwiftIDEUtilsTest/ClassificationTests.swift

```diff
@@ -121,4 +121,379 @@ public class ClassificationTests: XCTestCase {
+      ]
+    )
+  }
+  
+  public func testStringLikeLiterals() {
+    assertClassification(
+      "var s1 = \"abc абвгд あいうえお\""
+      , expected: [
+        ClassificationSpec(source: "var", kind: .keyword),
+        ClassificationSpec(source: "s1", kind: .identifier),
+        ClassificationSpec(source: "\"abc абвгд あいうえお\"", kind: .stringLiteral)
+      ]
+    )
+  }
+  
+  public func testOperatorDecl() {
+    assertClassification(
+      """
+      infix operator *-* : FunnyPrecedence
+      """
+      , expected: [
+        ClassificationSpec(source: "infix", kind: .keyword),
+        ClassificationSpec(source: "operator", kind: .keyword),
+        ClassificationSpec(source: "*-*", kind: .operatorIdentifier),
+        ClassificationSpec(source: "FunnyPrecedence", kind: .identifier)
+      ]
+    )
+  }
+  
+  public func testPrecedenceGroup() {
+    XCTExpectFailure("'left' is classified as a keyword in the test")
+    assertClassification(
+      """
+      precedencegroup FunnyPrecedence {
+        associativity: left
+        higherThan: MultiplicationPrecedence
+      }
+      """
+      , expected:  [
+        ClassificationSpec(source: "precedencegroup", kind: .keyword),
+        ClassificationSpec(source: "FunnyPrecedence", kind: .identifier),
+        ClassificationSpec(source: "associativity", kind: .keyword),
+        ClassificationSpec(source: "left", kind: .identifier),
+        ClassificationSpec(source: "higherThan", kind: .keyword),
+        ClassificationSpec(source: "MultiplicationPrecedence", kind: .identifier)
+      ]
+    )
+  }
+  
+  public func testInheritence() {
+    assertClassification(
+      "protocol Prot2 : Prot {}", expected: [
+        ClassificationSpec(source: "protocol", kind: .keyword),
+        ClassificationSpec(source: "Prot2", kind: .identifier),
+        ClassificationSpec(source: "Prot", kind: .typeIdentifier)
+      ]
+    )
+  }
+  
+  public func testGenericType() throws {
+    XCTExpectFailure(" '_' is not classified")
+    assertClassification(
+      "func genFn<T : Prot>(_: T) -> Int where T.Blarg : Prot2 {}"
+      , expected: [
+        ClassificationSpec(source: "func", kind: .keyword),
+        ClassificationSpec(source: "genFn", kind: .identifier),
+        ClassificationSpec(source: "T", kind: .identifier),
+        ClassificationSpec(source: "Prot", kind: .typeIdentifier),
+        ClassificationSpec(source: "_", kind: .keyword),
+        ClassificationSpec(source: "T", kind: .typeIdentifier),
+        ClassificationSpec(source: "Int", kind: .typeIdentifier),
+        ClassificationSpec(source: "where", kind: .keyword),
+        ClassificationSpec(source: "T", kind: .typeIdentifier),
+        ClassificationSpec(source: "Blarg", kind: .typeIdentifier),
+        ClassificationSpec(source: "Prot2", kind: .typeIdentifier)
+      ])
+  }
+  
+  public func testStringInterpolationAnchor() {
+    assertClassification(
+      "\"This is string \\(1) interpolation\""
```

Comment:
I think if you write this as a raw string `#"` or a multiline string, it becomes easier to read because you don’t need to do any escaping. 

Same in the `ClassificationSpec` below